src/runtime/entries/cloudflare-pages.ts

```diff
@@ -0,0 +1,52 @@
+import '#internal/nitro/virtual/polyfill'
+import { requestHasBody, useRequestBody } from '../utils'
+import { nitroApp } from '../app'
+
+export async function onRequest (context) {
+  // Contents of context object
+  const {
+    request, // same as existing Worker API
+    env, // same as existing Worker API
+    // params, // if filename includes [id] or [[path]]
+    // waitUntil, // same as ctx.waitUntil in existing Worker API
+    next // used for middleware or to fetch assets
+    // data, // arbitrary space for passing data between middlewares
+  } = context
+
+  try {
+    // const asset = await env.ASSETS.fetch(request, { cacheControl: assetsCacheControl })
+    const asset = await next()
+    if (asset.status !== 404) {
+      return asset
+    }
+  } catch (_err) {
+    // Ignore
+  }
+
+  const url = new URL(request.url)
+  let body
+  if (requestHasBody(request)) {
+    body = await useRequestBody(request)
+  }
+
+  const r = await nitroApp.localCall({
+    env,
+    context,
```

Comment:
FYI this currently has no effect ([src](https://github.com/unjs/unenv/blob/e389fb5d73af908c6e15a4c1a9dddd6770326e31/src/runtime/fetch/call.ts#L18)). unenv only picks known keys from input. But we should definitely add cf context in later changes.